crates/api/src/local_user/notifications/mark_comment_mention_read.rs

```diff
@@ -0,0 +1,50 @@
+use actix_web::web::{Data, Json};
+use lemmy_api_common::{
+  context::LemmyContext,
+  person::{MarkPersonCommentMentionAsRead, PersonCommentMentionResponse},
+};
+use lemmy_db_schema::{
+  source::person_comment_mention::{PersonCommentMention, PersonCommentMentionUpdateForm},
+  traits::Crud,
+};
+use lemmy_db_views::structs::LocalUserView;
+use lemmy_db_views_actor::structs::PersonCommentMentionView;
+use lemmy_utils::error::{LemmyErrorExt, LemmyErrorType, LemmyResult};
+
+#[tracing::instrument(skip(context))]
+pub async fn mark_comment_mention_as_read(
+  data: Json<MarkPersonCommentMentionAsRead>,
+  context: Data<LemmyContext>,
+  local_user_view: LocalUserView,
+) -> LemmyResult<Json<PersonCommentMentionResponse>> {
+  let person_comment_mention_id = data.person_comment_mention_id;
+  let read_person_comment_mention =
+    PersonCommentMention::read(&mut context.pool(), person_comment_mention_id).await?;
+
+  if local_user_view.person.id != read_person_comment_mention.recipient_id {
+    Err(LemmyErrorType::CouldntUpdateComment)?
+  }
+
+  let person_comment_mention_id = read_person_comment_mention.id;
+  let read = Some(data.read);
+  PersonCommentMention::update(
+    &mut context.pool(),
+    person_comment_mention_id,
+    &PersonCommentMentionUpdateForm { read },
+  )
+  .await
+  .with_lemmy_type(LemmyErrorType::CouldntUpdateComment)?;
+
+  let person_comment_mention_id = read_person_comment_mention.id;
+  let person_id = local_user_view.person.id;
+  let person_comment_mention_view = PersonCommentMentionView::read(
+    &mut context.pool(),
+    person_comment_mention_id,
+    Some(person_id),
+  )
+  .await?;
+
+  Ok(Json(PersonCommentMentionResponse {
+    person_comment_mention_view,
+  }))
```

Comment:
How about changing the return values of mark as read to simply SuccessResponse? Then it would also make sense to switch to a single mark_as_read endpoint so clients dont have to differentiate between comment, post, etc.

Reply:
On updates and creates, its really helpful to get back the changed / new item. Then the front end can replace that object after a success, and make the findAndReplace function generic, instead of having a bunch of specific ones for each action, where the response tells you nothing about what changed.